app/dao/notifications_dao.py

```diff
@@ -107,7 +108,11 @@ def bulk_insert_notifications(notifications):
             notification.status = NOTIFICATION_CREATED
 
     # TODO: Add error handling (Redis queue?) for failed notifications
-    return db.session.bulk_save_objects(notifications)
+    try:
+        return db.session.bulk_save_objects(notifications)
+    except IntegrityError as e:
+        current_app.logger.error(e)
+        raise
```

Comment:
I don't think to be logged here if we raise it again: the higher layers will catch it and log. If we have it this way, that would log it twice in the logs.

Reply:
I am not sure I get it, can you help me understand why we catch an exception that we throw with no additional behavior? This is similar to how the code was before but with more lines.

I think we had this conversation 2-3 months ago. I would ask the same that I asked back then: can you identify best practices in the Python community that encourages this practice?